solutions/USACO feb17-friendcross.mdx

```diff
@@ -396,4 +396,75 @@ int main() {
 
 ## Solution 3
 
-After generating the appropriate sequence of updates and queries, can apply divide & conquer as described in the module.
\ No newline at end of file
+After generating the appropriate sequence of updates and queries, can apply divide & conquer as described in the module.
+
+## Solution 4
+
+Let us consider an easier version: where we're just asked to count number of crossing pairs (ignoring whether the cows are friendly or not). To calculate the number of crossing pairs, we can add the cows in road 1 in reverse, while counting the number of cows that are added before in road 2. We can use a binary indexed tree to count this. This takes $\mathcal O(N)$ space and $\mathcal O(N\log N)$ time.
```

Comment:
> Let us consider an easier version: where we're just asked to count number of crossing pairs (ignoring whether the cows are friendly or not).

Let us consider an easier version; suppose that we just want to count the number of crossing pairs (ignoring whether the cows are friendly or not).